lib/segment/src/index/field_index/map_index/mod.rs

```diff
@@ -645,47 +756,90 @@ mod tests {
     use std::collections::HashSet;
     use std::path::Path;
 
+    use rstest::rstest;
     use tempfile::Builder;
 
     use super::*;
     use crate::common::rocksdb_wrapper::open_db_with_existing_cf;
 
     const FIELD_NAME: &str = "test";
 
-    fn save_map_index<N: MapIndexKey + ?Sized>(data: &[Vec<N::Owned>], path: &Path)
-    where
+    #[derive(Clone, Copy)]
+    enum IndexType {
+        Mutable,
+        Immutable,
+        Mmap,
+    }
+
+    fn save_map_index<N>(
+        data: &[Vec<N::Owned>],
+        path: &Path,
+        index_type: IndexType,
+        into_value: impl Fn(&N::Owned) -> Value,
+    ) where
+        N: MapIndexKey + MmapValue + ?Sized,
         MapIndex<N>: PayloadFieldIndex + ValueIndexer,
+        <MapIndex<N> as ValueIndexer>::ValueType: Into<N::Owned>,
```

Comment:
A lot of type conversion happen here... Like you're using `String` and `SmolStr` at the same time. Maybe you could use just one of them?

Reply:
There is a conversion chain: `Value -> ValueType -> N::Owned`. `Value` is json, `ValueType` is `String`, and `N::Owned` is `SmolStr`. We use `SmolStr` here to avoid a lot of conversion while retrieve values and checks